distros/spring/src/test/java/com/connexta/transformation/service/TransformControllerTest.java

```diff
@@ -66,15 +69,64 @@ public void testAcceptedResponseEntity() throws Exception {
                             .finalLocation(new URI(TEST_URI))
                             .metacardLocation(new URI(TEST_URI))))
                 .characterEncoding(StandardCharsets.UTF_8.name()))
-        .andExpect(status().isAccepted());
+        .andExpect(status().isCreated())
+        .andExpect(header().string("Location", "http://localhost/transform/id123"))
+        .andExpect(redirectedUrl("http://localhost/transform/id123"));
+  }
+
+  @Test
+  public void testCurrentLocationCannotBeNull400() throws Exception {
+    this.mockMvc
+        .perform(
+            post("/transform")
+                .header(ACCEPT_VERSION, ACCEPT_VERSION_NUM)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(
+                    mapper.writeValueAsString(
+                        new TransformRequest()
+                            .currentLocation(null)
+                            .finalLocation(new URI(TEST_URI))
+                            .metacardLocation(new URI(TEST_URI))))
+                .characterEncoding(StandardCharsets.UTF_8.name()))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  public void testFinalLocationNullCannotBe400() throws Exception {
+    this.mockMvc
+        .perform(
+            post("/transform")
+                .header(ACCEPT_VERSION, ACCEPT_VERSION_NUM)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(
+                    mapper.writeValueAsString(
+                        new TransformRequest()
+                            .currentLocation(new URI(TEST_URI))
+                            .finalLocation(null)
+                            .metacardLocation(new URI(TEST_URI))))
+                .characterEncoding(StandardCharsets.UTF_8.name()))
+        .andExpect(status().isBadRequest());
   }
 
-  @Test(expected = ResourceAccessException.class)
-  public void testInvalidServiceUrl() {
-    String badServiceUrl = "http://bad.url";
-    TransformController controller =
-        new TransformController(new RestTemplateBuilder(), badServiceUrl, tManager);
+  @Test
+  public void testMetacardlLocationCannotBeNull400() throws Exception {
+    this.mockMvc
+        .perform(
+            post("/transform")
+                .header(ACCEPT_VERSION, ACCEPT_VERSION_NUM)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(
+                    mapper.writeValueAsString(
+                        new TransformRequest()
+                            .currentLocation(new URI(TEST_URI))
+                            .finalLocation(new URI(TEST_URI))
+                            .metacardLocation(null)))
+                .characterEncoding(StandardCharsets.UTF_8.name()))
+        .andExpect(status().isBadRequest());
+  }
 
-    controller.transform(ACCEPT_VERSION, new TransformRequest());
+  @Test(expected = NullPointerException.class)
+  public void testManagerCannotBeNull() {
+    new TransformController(null);
   }
```

Comment:
❗️ I would expect to add a lot more test cases that validates some of the restrictions defined on the model for the body passed in for the request.